### DISCUSSION TITLE
Is the exist env able to add more robots inside?

### DISCUSSION THREAD

**Discussion by ashley1223** [jr-robotics/robo-gym#30] [Q&A]
Hi hope all well!
I and a new study student start to multi robot system with ur5, and feel robo-gym is quite nice!
But may I ask, if I want to  add more ur5 robots there (more than one) into EndEffectorPositioningURSim Env, is that possible? And also if i want to add something new such as conveyor, parts on conveyor, is that workable? Thank you!

**Comment by matteolucchi** [comment]
HI @ashley1223 ! Thank you for your interest in robo-gym!

>  if I want to add more ur5 robots there (more than one) into EndEffectorPositioningURSim Env, is that possible? 

Yes, it is possible but I don't know how much effort that requires. This is not a limitation of robo-gym itself but rather of the UR Gazebo simulation, it is possible to add multiple robots to the same Gazebo simulation and for sure you can find examples of that around. The two main concerns that I know of are:
- When spawning 2 robots with the same URDF model there might be conflicts with names of robot links, to solve this namespaces have to be setup. It could be that is already implemented in the [UR Simulation](https://github.com/jr-robotics/universal_robot) but you'll have to check that. 
- The simulation might become very heavy when simulating 2 robots at the same time, but I guess this is true for all simulation softwares 

> And also if i want to add something new such as conveyor, parts on conveyor, is that workable? 

Again here the answer is: it depends on Gazebo, so if you can find a Gazebo simulation of a conveyor belt or if you design your own than for sure this can be integrated in robo-gym.

As a general idea, robo-gym is currently based on Gazebo for simulation, so whatever can be simulated in Gazebo can be integrated in robo-gym. 

I hope this answers your questions :) Let me know if you have more doubts
